// @react95/core/cjs/shared/events.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
class Emitter {
  constructor() {
    this.listeners = {};
  }
  on(eventName, callback) {
    if (!this.listeners[eventName]) {
      this.listeners[eventName] = [];
    }
    this.listeners[eventName].push(callback);
  }
  off(eventName, callback) {
    if (this.listeners[eventName]) {
      this.listeners[eventName] = this.listeners[eventName].filter(
        (cb) => cb !== callback
      );
    }
  }
  emit(eventName, data) {
    if (this.listeners[eventName]) {
      this.listeners[eventName].forEach((callback) => callback(data));
    }
  }
}
var ModalEvents = /* @__PURE__ */ ((ModalEvents2) => {
  ModalEvents2["AddModal"] = "add-modal";
  ModalEvents2["RemoveModal"] = "remove-modal";
  ModalEvents2["ModalVisibilityChanged"] = "modal-visibility-changed";
  ModalEvents2["MinimizeModal"] = "minimize-modal";
  ModalEvents2["RestoreModal"] = "restore-modal";
  return ModalEvents2;
})(ModalEvents || {});
const modals = new Emitter();
exports.Emitter = Emitter;
exports.ModalEvents = ModalEvents;
exports.modals = modals;
